### PR TITLE
v4 — Change .has-error to .has-danger for uniformity.

### PR DESCRIPTION
**Pull request by kkirsche**
X-Ref: Validation and states #17021

### CODE REVIEW

**Comment by zacechola** [comment]
X-ref:  #17141

And also any other instances of `.has-error` in the documentation.

**Comment by kkirsche** [comment]
@zacechola This replaces every instance except in the `dist` and `docs/dist` folder excluding pending PRs

**Comment by zacechola** [comment]
ok, cool. thanks!

**Comment by kkirsche** [comment]
Course :+1: — Thank find and replace in Sublime Text :wink:

**Comment by ADmad** [comment]
IMHO "error" is a better choice of word than "danger" and I would suggest doing the opposite: rename "danger" to "error" in all class names.

**Comment by zacechola** [comment]
@ADmad That wouldn't make sense in many situations you're likely to use red colors, such as on a delete button, where the context has nothing to do with errors.

**Comment by cvrebert** [comment]
I try not to indulge in this, but I'll make an exception this one time:
ec699c7ecf558a6cb9f5e6ef2f3ba3b2

**Comment by ADmad** [comment]
@zacechola Okay I see your point, but in the context of validation errors using class named "error" makes more sense. So let's stick to context/semantics and keep this class name unchanged. It seems the reasoning behind proposed name change is that it uses same color as the "danger" classes. If you want to have uniformity based on color rename them all to "red".

@cvrebert The creator of that meme needs to watch https://www.youtube.com/watch?v=-FdKPEA17m4 :stuck_out_tongue:

File: scss/_forms.scss [hunk 1]
```diff
@@ -288,7 +288,7 @@ input[type="checkbox"] {
   }
 }
 
-.has-error {
+.has-danger {
   @include form-control-validation(error, $brand-danger);
```
Comment (cvrebert) [inline scss/_forms.scss]: This should also be "danger" instead of "error".
(Removing that parameter altogether is a matter for another issue/PR, #17323)

Comment (kkirsche) [inline scss/_forms.scss]: Fixed @cvrebert

**Comment by kkirsche** [comment]
Rebased against master to clear up merge conflicts. Wanted to follow up here and see if there was anything needed from me to keep this moving forward?

**Comment by cvrebert** [comment]
CC: @mdo for review

**Pull request merged**